packages/edit-post/src/hooks/use-post-history.js

```diff
@@ -35,6 +35,13 @@ export default function usePostHistory( initialPostId, initialPostType ) {
 		[ { postId: initialPostId, postType: initialPostType } ]
 	);
 
+	const editPostTypeProps = useMemo( () => {
```

Comment:
@noisysocks How's this vibe? 😄 

Stores the initial post id and type when the post editor loads and shoves it in the settings.

Could be `__experimental` too.

Reply:
We're not exporting the `usePostHistory` hook anywhere, so I reckon we probably don't need the `__experimental` prefix?

In terms of the `useMemo` call, is that mostly here to ensure that `editPostTypeProps` isn't a new object on every render?

Reply:
> In terms of the useMemo call, is that mostly here to ensure that editPostTypeProps isn't a new object on every render?


Thank you for testing @andrewserong !!

That was the idea, but I'm not so sure it does much given that the hook is called in the editor only once as far as I know 🤔 

I can test this.

Reply:
> We're not exporting the usePostHistory hook anywhere, so I reckon we probably don't need the __experimental prefix?


Oh, wanted to say that I mentioned `__experimental` because it's a new property on the editorSettings object, which is available as a selector everywhere.

Reply:
Ah, gotcha 👍

Reply:
`@wordpress/editor` shouldn't "know" anything about `@wordpress/edit-post`. This violates that by having a setting in `getEditorSettings()` that is specific to `@wordpress/edit-post`.

Reply:
I think it's matter of semantics. Let's consider changing the var name so that it doesn't explicitly refer to `editPost`.

```js
	const initialPost = {
		type: initialPostType,
		id: initialPostId,
	};
```

Reply:
Yep that's much better. Still don't think we should put it in editor settings though. `FullscreenModeClose` is a descendent of `Editor` in the react tree so better imo to just pass `initialPost` down via props. That way the data never leaves `edit-post`.